### PR TITLE
Document module `io.micronaut:micronaut-jackson-databind`

### LINKED ISSUE #9531: Document module
### Issue description

We have to document the module `io.micronaut:micronaut-jackson-databind` and the usage of JsonMapper (which is implementation agnostic instead of ObjectMapper (micronaut-jackson-databind or micronaut-serde-jackson)

### PR DESCRIPTION
**Pull request by wetted**
closes #9531

### CODE REVIEW

**Comment by wetted** [comment]
@sdelamo I'm kinda unsure what all to say about this if anything else, or if it's even in the right place. Happy for further guidance.

**Comment by sdelamo** [comment]
> @sdelamo I'm kinda unsure what all to say about this if anything else, or if it's even in the right place. Happy for further guidance.

I have rewritten the PR slightly

File: src/main/docs/guide/httpServer/jsonBinding.adoc [hunk 1]
```diff
@@ -4,9 +4,30 @@
 
 The most common data interchange format nowadays is JSON.
 
-In fact, the defaults in the api:http.annotation.Controller[] annotation specify that the controllers in Micronaut framework consume and produce JSON by default.
+By default, the api:http.annotation.Controller[] annotation specifies that the controllers in Micronaut framework consume and produce JSON by default.
 
-To do so in a non-blocking manner, Micronaut framework builds on the https://github.com/FasterXML/jackson[Jackson] Asynchronous JSON parsing API and Netty, such that the reading of incoming JSON is done in a non-blocking manner.
+Since Micronaut Framework 4.0, users must choose how they want to serialize (Jackson Databind or Micronaut Serialization). Both approaches allow the usage of https://micronaut-projects.github.io/micronaut-serialization/snapshot/guide/index.html#jacksonAnnotations[Jackson Annotations].
```
Comment (graemerocher) [inline src/main/docs/guide/httpServer/jsonBinding.adoc]: don't link to `snapshot` link to `latest`

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: src/main/docs/guide/httpServer/jsonBinding.adoc (hunk 1 above)
Comment (sdelamo) [reply src/main/docs/guide/httpServer/jsonBinding.adoc]: resolved via https://github.com/micronaut-projects/micronaut-core/pull/9541/commits/0da3905083bfc76340d9d01310a5b1ceec2056b4

**Review by graemerocher — APPROVED** [review]

**Pull request merged**